spikes/stl_cxxsupport/main.cpp

```diff
@@ -94,5 +95,20 @@ auto main() -> int
 	auto back = span0.back();
 	log_info("span0.back() = %i", back);
 
+	log_info("Test std::range features");
+
+	constexpr auto range = [](const auto a, const auto b) {
+		auto x = std::ranges::views::iota(a) | std::ranges::views::take(b - a + 1);
+		return std::ranges::views::all(x);
+	};
+
+	for (auto x: range(1, 10)) {
+		log_info("x: %i", x);
+	}
```

Comment:
This will prompt 

```
000:00:02.001 [INFO] [main.cpp:106] main > x: 1
000:00:02.001 [INFO] [main.cpp:106] main > x: 2
000:00:02.001 [INFO] [main.cpp:106] main > x: 3
000:00:02.001 [INFO] [main.cpp:106] main > x: 4
000:00:02.001 [INFO] [main.cpp:106] main > x: 5
000:00:02.001 [INFO] [main.cpp:106] main > x: 6
000:00:02.001 [INFO] [main.cpp:106] main > x: 7
000:00:02.001 [INFO] [main.cpp:106] main > x: 8
000:00:02.001 [INFO] [main.cpp:106] main > x: 9
000:00:02.001 [INFO] [main.cpp:106] main > x: 10
```

**Note.** For some usage of range, last value is excluded. Using math notation `range(a,b`) is equivalent to [a, b[.

It make sense for a for-loop of n values, range(n) should repeat n-times, from 0 to n-1.

Reply:
Can you explain a use case to figure out what is the best practice here? (Consider or not to include last value)